src/completion_queue.rs

```diff
@@ -46,18 +47,18 @@ impl CompletionQueue {
     }
 
     /// Request notification on next complete event arrive
-    pub(crate) fn req_notify(&self, solicited_only: bool) -> io::Result<()> {
+    pub fn req_notify(&self, solicited_only: bool) -> io::Result<()> {
         let errno = unsafe {
             ibv_req_notify_cq(self.inner_cq.as_ptr(), if solicited_only { 1 } else { 0 })
         };
         if errno != 0_i32 {
-            return Err(io::Error::from_raw_os_error(errno));
+            return Err(io::Error::from_raw_os_error(errno.mul(-1_i32)));
```

Comment:
is `0 - errno`  better?

Reply:
https://github.com/datenlord/async-rdma/pull/28/commits/61644b432d838cfa16feb63b2e17b222d4125b17 fixed